pdp-info/info.go

```diff
@@ -0,0 +1,150 @@
+package info
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type streamObj struct {
+	stream PipService_GetRequestsStreamServer
+	err    chan error
+}
+
+type streamsPool struct {
+	streams map[string][]streamObj
+	sync.RWMutex
+}
+
+func newStreamsPool() *streamsPool {
+	return &streamsPool{
+		streams: make(map[string][]streamObj),
+	}
+}
+
+func (p *streamsPool) RegisterStream(stream PipService_GetRequestsStreamServer) chan error {
+	cerr := make(chan error, 1)
+
+	resp, err := stream.Recv()
+	if err != nil {
+		cerr <- err
+		return cerr
+	}
+
+	inAttrs := resp.GetAttributes()
+	if len(inAttrs) != 1 {
+		cerr <- fmt.Errorf("Expected exactly one attribute, got: %d", len(inAttrs))
+		return cerr
+	}
+
+	name := inAttrs[0].GetValue()
```

Comment:
it is not explicit here.  maybe let PIP send some string like "registration" to explicitly show the intention.

Reply:
Do you mean to add verify GetId() that should be 'registration' (ATM is used id 'ident')?